### ISSUE TITLE
sigmoid approximation and Straight-Through Estimator: SPAN in leaf_dynamics

### ISSUE THREAD

**Issue by SarahAlidoost** [WUR-AI/diffWOFOST#68]
While working on #66, I noticed that in `leaf_dynamics`, the PyTorch gradient of the parameter `SPAN` with respect to `LAI` is zero. This means the optimization loop can’t update `SPAN`. This didn’t happen in the release branch. Here I explain why and suggest a possible solution.

In the original `leaf_dynamics`, `SPAN` is a hard threshold. In the differentiable version, we replaced it with a sigmoid approximation with a “sharpness” parameter. In the release branch, the sharpness was set to 1000. But in [this commit](https://github.com/WUR-AI/diffWOFOST/pull/54/changes#diff-439253bedd396445b48656a2016165f4fb2764fe7bfeb49f251c5e5f2ac827df), it was changed to a very small value, 1e-16, to ensure the model outputs exactly match all the test data. The problem is that such a tiny sharpness makes the sigmoid behave almost like a step function, which results in zero gradients. This illustrates the classic trade-off: approximations that allow gradients to exist can conflict with perfectly matching test values (also raise in #24)

There are a few ways to fix this. One approach is the [**Straight-Through Estimator (STE)**](https://apxml.com/courses/practical-llm-quantization/chapter-4-quantization-aware-training-qat/straight-through-estimator-ste) method, also see references below. In short, we use the hard threshold in the forward pass (like the original version) but compute gradients using the sigmoid in the backward pass. This allows optimization to proceed while keeping outputs close to the original behavior. This is implemented for leaf_dynamics, see [here](https://github.com/WUR-AI/diffWOFOST/blob/3fbf6d7a78f3a543068be632f216a4e4f5710caa/src/diffwofost/physical_models/crop/leaf_dynamics.py#L287-L312).  

ref: [Understanding Straight-Through Estimator in Training Activation Quantized Neural Nets](https://doi.org/10.48550/arXiv.1903.05662)
